tensorflow/stream_executor/cuda/cuda_gpu_executor.cc

```diff
@@ -1129,6 +1129,30 @@ DeviceDescription *CUDAExecutor::PopulateDeviceDescription() const {
           CU_DEVICE_ATTRIBUTE_MAX_REGISTERS_PER_MULTIPROCESSOR, device_)
           .ValueOrDie());
 
+  const char* blank_ptx =
```

Comment:
I think this deserves a comment explaining why the heck we're doing this thing.  :)

Reply:
The reasoning behind it is that CUDA's occupancy calculator functions are the best(/only?) way to access the maximum number of blocks per SM, since that parameter can't be queried otherwise.  In order to call these functions we need to provide a kernel, and since it doesn't appear that we have access to the actual kernel when XLA's partition_assignment.cc attempts to set the number of threads per block, providing a dummy kernel seemed a quick alternative.

Ideally, XLA would be calling the occupancy calculator when setting launch parameters, but based on my (limited) understanding of the code path, it doesn't appear that the partition_assignment code has enough information to make this call. (Namely, access to the kernel)

Reply:
(Understood, I just meant that we should explain this to readers with a comment in the code.)

Reply:
Ah, I see, that kind of comment :)